File: Metode avansate de Programare/Teme/familydoctor/saptamana 1-7/CSharp/FamilyDoctor/FamilyDoctor/Domain/Patient.cs
using System;
using System.Collections.Generic;

namespace FamilyDoctor.Domain
{
    public class Patient
    {
        private string id, name;
        private int age;
        List<string> appointmentIDs;

        public string ID { get { return id; } set { id = value; } }
        public string Name { get { return name; } set { name = value; } }
        public int Age { get { return age; } set { age = value; } }
        public List<string> AppointmentIDs { get { return appointmentIDs; } }
        public int AppointmentsCount { get { return appointmentIDs.Count; } }

        public Patient()
        {
            appointmentIDs = new List<string>();
        }

        public Patient(string id, string name, int age)
        {
            this.id = id;
            this.name = name;
            this.age = age;

            appointmentIDs = new List<string>();

            Validate();
        }

        public void Validate()
        {
            if (id == null || id == string.Empty)
                throw new ValidationException("Invalid ID");

            if (name == null || name == string.Empty)
                throw new ValidationException("Invalid name");

            if (age < 0)
                throw new ValidationException("Invalid age");
        }

        public override bool Equals(object other)
        {
            if (other == null)
                return false;

            if (other.GetType() == typeof(Patient))
                if (id == ((Patient)other).ID)
                    return true;
            return false;
        }

        public override string ToString()
        {
            return name + " (ID:" + id + "; Age:" + age + ")";
        }
    }
}

File: Metode avansate de Programare/Teme/familydoctor/saptamana 1-7/Java/MedicDeFamilie/src/familydoctor/model/Patient.java
package familydoctor.model;

import java.util.List;

public class Patient {
    private String id, name;
    private int age;
    private List<Appointment> appointments;

    public Patient() {

    }

    public Patient(String id, String name, int age) {
        this.id = id;
        this.name = name;
        this.age = age;

        validate();
    }

    public String getId() { return id; }
    public void setId(String value) {
        id = value;
    }

    public String getName() {
        return name;
    }
    public void setName(String value) {
        name = value;
    }

    public int getAge() {
        return age;
    }
    public void setAge(int value) {
        age = value;
    }

    public void validate()
    {
        if (id == null || id == "")
            throw new ValidationException("Invalid ID");

        if (name == null || name == "")
            throw new ValidationException("Invalid name");

        if (age < 0)
            throw new ValidationException("Invalid age");
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;

        if (obj == null)
            return false;

        if (getClass() != obj.getClass())
            return false;

        Patient other = (Patient) obj;
        if (id == null) {
            if (other.id != null)
                return false;
        } else if (!id.equals(other.id))
            return false;

        return true;
    }



}